zerver/lib/upload.py

```diff
@@ -184,21 +183,19 @@ def upload_image_to_s3(
 
     key.set_contents_from_string(contents, headers=headers)  # type: ignore # https://github.com/python/typeshed/issues/1552
 
-def get_total_uploads_size_for_user(user: UserProfile) -> int:
-    uploads = Attachment.objects.filter(owner=user)
-    total_quota = uploads.aggregate(Sum('size'))['size__sum']
-
-    # In case user has no uploads
-    if (total_quota is None):
-        total_quota = 0
-    return total_quota
-
-def within_upload_quota(user: UserProfile, uploaded_file_size: int) -> bool:
-    total_quota = get_total_uploads_size_for_user(user)
-    if (total_quota + uploaded_file_size > user.quota):
-        return False
-    else:
-        return True
+def currently_used_upload_space(realm: Realm) -> int:
+    used_space = Attachment.objects.filter(realm=realm).aggregate(Sum('size'))['size__sum']
+    # In case realm has no uploads
+    if (used_space is None):
+        return 0
+    return used_space
+
+def check_upload_within_quota(realm: Realm, uploaded_file_size: int) -> None:
+    if realm.uploads_quota_gb is None:
```

Comment:
Hmm. How about making `uploads_quota_bytes` return None if `uploads_quota_gb` is None? Then this line can refer to `uploads_quota_bytes` just like the line below it, which simplifies thinking about this code.